test/all-options.test.js

```diff
@@ -311,13 +325,14 @@ describe('when applied with all options', () => {
               compilationEventBindings = chunkPluginEnvironment.getEventBindings();
             });
 
-            it('should get all warnings', () => {
+            it('should get all warnings', (done) => {
```

Comment:
Remove, because this is a synchronous test (ensured via `PluginEnvironment`). This should not use a callback. or return a promise. This will also help to make sure the `parallel` feature being added does not break this expected behavior.